operator/config.go

```diff
@@ -5,80 +5,61 @@ package operator
 import (
 	"encoding/json"
 	"fmt"
-
-	"github.com/observiq/stanza/database"
-	"go.uber.org/zap"
 )
 
 // Config is the configuration of an operator
 type Config struct {
-	Builder
+	MultiBuilder
 }
 
-// Builder is an entity that can build operators
-type Builder interface {
+type Identifier interface {
 	ID() string
 	Type() string
-	Build(BuildContext) (Operator, error)
-	SetNamespace(namespace string, exclude ...string)
 }
 
-// BuildContext supplies contextual resources when building an operator.
-type BuildContext struct {
-	Database   database.Database
-	Parameters map[string]interface{}
-	Logger     *zap.SugaredLogger
-}
-
-// registry is a global registry of operator types to operator builders.
-var registry = make(map[string]func() Builder)
-
-// Register will register a function to an operator type.
-// This function will return a builder for the supplied type.
-func Register(operatorType string, newBuilder func() Builder) {
-	registry[operatorType] = newBuilder
+// Builder is an entity that can build operators
+type Builder interface {
+	Identifier
+	Build(BuildContext) (Operator, error)
 }
 
-// IsDefined will return a boolean indicating if an operator type is registered and defined.
-func IsDefined(operatorType string) bool {
-	_, ok := registry[operatorType]
-	return ok
+// Builder is an entity that can build operators
+type MultiBuilder interface {
+	Identifier
+	BuildMulti(BuildContext) ([]Operator, error)
```

Comment:
Can you comment on the need for `MultiBuilder` as opposed to the previous implementation? I'm assuming it has to do with avoiding sequential dependencies between operators, or something to that effect, but I'd like to understand more concretely.

Reply:
`MultiBuilder` is basically just an extension of the `Builder` interface that can build multiple operators from a single config rather than only allowing building a single operator. This way we can treat plugin configs and operator configs exactly the same. 

Now, both operator blocks and plugin blocks are unmarshaled into a struct that implements `MultiBuilder`, and when built, plugins configs turn into a slice of many operators (one per operator in the plugin pipeline), and operator configs turn into a slice of a single operator (the operator that the `Builder` interface returns).